flink-runtime/src/main/java/org/apache/flink/runtime/shuffle/NettyShuffleDescriptor.java

```diff
@@ -57,6 +58,11 @@ public ResultPartitionID getResultPartitionID() {
 		return resultPartitionID;
 	}
 
+	@Override
+	public Optional<ResourceID> hasLocalResources() {
```

Comment:
you know, whenever I see this method I always first assume that it returns a boolean.

Reply:
maybe we could rename it to `storesLocalResourcesOn()`

Reply:
agree, it might bring confusing here.

Reply:
True, `storesLocalResourcesOn` fits better